### PR TITLE
[configuration] Make cmake configuration fail early upon boost or openBLAS installation failure

### PR DESCRIPTION
**Pull request by lambdaxdotx**
**TLDR:** This PR introduces a couple of checks for the `cmake` configuration process to make it fail upon the installation failure of either `boost` or `openblas`.

## Problem

At the moment, whenever any `tools/download_*.sh` script fails, the configuration process continues w/o any explicit message, for failing only at the end. This may surprise and mislead the (non-dev) user: issue https://github.com/NeuralNetworkVerification/Marabou/issues/481 seems to be an example of that.

Catching the erratic state of the configuration and fail early seems a better strategy to me.

## Proposed solution

- Make `tools/download_*.sh` exit on any failure by passing the `-e` to `/bin/bash`
- Check the exit code of those scripts in the `CMakeLists.txt` and fail accordingly.

Let me know what you think.

### CODE REVIEW

**Comment by lambdaxdotx** [comment]
The CI fails because the `download_boost.sh` fails at some point. In particular, this means that the script was _already_ failing.

The next step, I guess, is to look at the precise command that fails in the script and let maintainers decide how to proceed.

**Comment by AleksandarZeljic** [comment]
Thanks for submitting the PR. I am pretty sure that the failure has nothing to do with the PR. A recent PR was pushed into master after passing CI checks, but once integrated failed because boost wasn't downloaded correctly only to suddenly work again without intervention. I will look into this.

**Comment by lambdaxdotx** [comment]
Hi @AleksandarZeljic

Thanks for giving this PR some attention. Let me know if you need help.
